blueprints/networking/glb-hybrid-neg-internal/variables.tf

```diff
@@ -0,0 +1,111 @@
+/**
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+variable "prefix" {
+  description = "Prefix used for resource names."
+  type        = string
+  validation {
+    condition     = var.prefix != ""
+    error_message = "Prefix cannot be empty."
+  }
+}
+
+variable "project_names" {
+  description = "The project names."
+  type = object({
+    landing  = string
+    spoke_01 = string
+  })
+  default = {
+    landing  = "landing"
+    spoke_01 = "spoke-01"
+  }
+}
+
+variable "projects_create" {
+  description = "Parameters for the creation of the new project."
+  type = object({
+    billing_account_id = string
+    parent             = string
+  })
+  default = null
+}
+
+variable "region_configs" {
+  description = "The primary and secondary region parameters."
+  type = object({
+    r1 = object({
+      region_name = string
+      zone        = string
+    })
+    r2 = object({
+      region_name = string
+      zone        = string
+    })
+  })
+  default = {
+    r1 = {
+      region_name = "europe-west1"
+      zone        = "europe-west1-b"
+    }
+    r2 = {
+      region_name = "europe-west2"
+      zone        = "europe-west2-b"
+    }
+  }
+}
+
+variable "test_vms_behind_ilb" {
+  description = "Whether there should be an ILB L4 in front of the test VMs in the spoke."
+  type        = string
+  default     = false
+}
+
+variable "vpc_landing_trusted_config" {
```

Comment:
I would group all vpc configurations under a single variable (`ip_config`?) so all IP ranges are specified together.

Reply:
Even better, I would add use factories for subnets and firewall rules, and use the same variable to point to the data folder as we use in FAST. WDYT?

Reply:
@ludoo I merged anything into one flat variable and added optionals so a user can override singularly any of those CIDRs.
If you don't mind, I'd refrain for now from using factories in this context, as I see them as a way to just replicate variable values in multiple places (i.e. regions, cidrs) without giving us too many advantages.